domain/src/main/java/org/oppia/android/domain/profile/ProfileManagementController.kt

```diff
@@ -721,8 +721,29 @@ class ProfileManagementController @Inject constructor(
     return imageFile.absolutePath
   }
 
-  private fun onlyLetters(name: String): Boolean {
-    return name.matches(Regex("^[ A-Za-z]+\$"))
+  private fun nameAllowed(name: String): Boolean {
+    return (
+      notEmptyNoSpacesAndContainsLetters(name) &&
+        noNumbers(name) &&
+        noSymbols(name) &&
+        noRepeatedUseOfAllowedSymbols(name)
+      )
+  }
+
+  private fun notEmptyNoSpacesAndContainsLetters(name: String): Boolean {
+    return name.matches(Regex("^.[\\w\\u00BF-\\u1FFF\\u2C00-\\uD7FF.'\\-]+\$"))
+  }
+
+  private fun noNumbers(name: String): Boolean {
+    return name.matches(Regex("^[^0-9]*\$"))
```

Comment:
You could also do: ``name.none { it.isDigit() }`` which might actually be more performant than the regex check.